query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -408,15 +407,15 @@ class AvgOverTimeFunction(var sum: Double = Double.NaN, var count: Int = 0) exte
   }
 }
 
-abstract class AvgOverTimeChunkedFunction(var sum: Double = Double.NaN, var count: Double = 0)
+abstract class AvgOverTimeChunkedFunction(var sum: Double = Double.NaN, var count: Int = 0)
   extends ChunkedRangeFunction[TransientRow] {
   override final def reset(): Unit = {
-    sum = Double.NaN;
-    count = 0d
+    sum = Double.NaN
+    count = 0
   }
 
   final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
-    sampleToEmit.setValues(endTimestamp, if (count > 0) sum/count else if (sum.isNaN()) sum else 0d)
+    sampleToEmit.setValues(endTimestamp, sum / count)
```

Comment:
I think it's still safer to have the divide by zero check here.  I see that in reset() the count is still 0, so there is still this possibility.

Reply:
yes, but Double.NaN/0 will always be double.NaN. Also, it's not possible for sum to be some value and count to be 0.

Reply:
also, when count in reset() is 0, sum is Double.NaN. So, Double.NaN/0 = Double.NaN.

Reply:
Still you want the change?

Reply:
We have a TODO to represent no-value using a special NaN instead of regular NaN. We have to check if operations on that NaN result in the same NaN. Given this, it may be better to retain the explicit check.

Reply:
Added the check @vishramachandran.

Reply:
We can't rely on NaN.  It is possible, if it is an Integer encoded column, to end up with a sum of 0 and count of 0.    Anyways will have a look.